### DISCUSSION TITLE
Adding a button to Dialog toolbar after afterLoadEntity

### DISCUSSION THREAD

**Discussion by DGoertzen** [serenity-is/Serenity#6762] [Help]
Is there a simple way within the framework to add buttons to the Dialog Toolbar based on data that is received from the Entity being loaded?  My understanding is that the Entity data is received after the call to initToolbar, so it doesn't work to override getToolbarButtons, as when getToolbarButtons is called the data that is used to determine which buttons to add (in my application) is not available yet.

I understand that I can do this using JQuery, but I wonder if there is a way to do it that is supported by the Serenity framework.

Thanks for any help!

**Comment by LSNicholls** [comment]
I have exactly the same problem in multiple places.  This resolution may not work for you, depending on the scenario, but here is what I am doing:

1. put the potential buttons in during getToolbarButtons()
2. in updateInterface(), or sometimes afterLoadEntity(), depending on whether the adjustments are user-centric or entity-centric, check the data and determine what I need to do, adjusting them (using this.toolbar.findButton).  In some cases I set them readonly at that point or add a class to disable them, in others I use .remove().

It seems a bit a**-backwards, but I couldn't find a better way.
